### PR TITLE
[SYCLomatic] Include cmath when meet math API from standard library

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/test/dpct/implicit_include.cu [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// RUN: dpct -out-root %T/implicit_include %s --cuda-include-path="%cuda-path/include"
+// RUN: FileCheck --input-file %T/implicit_include/implicit_include.dp.cpp --match-full-lines %s
+
+// CHECK: #include <cmath>
```
Comment (zhimingwang36) [inline clang/test/dpct/implicit_include.cu:4]: if "#include <cmath>" already there,  will the PR insert another "#include <cmath>",  if yes, we need further refine.

**Review by zhimingwang36 — CHANGES_REQUESTED** [review]
LGTM

File: clang/test/dpct/implicit_include.cu (hunk 1 above)
Comment (zhiweij1) [reply clang/test/dpct/implicit_include.cu:4]: I checked our code, we already supported this case.
I added a new lit case to test it.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**